packages/@aws-cdk/aws-kinesisfirehose-alpha/test/integ.delivery-stream.source-stream.ts

```diff
@@ -43,4 +50,30 @@ new firehose.DeliveryStream(stack, 'Delivery Stream No Source Or Encryption Key'
   destination: mockS3Destination,
 });
 
-app.synth();
+const integTest = new IntegTest(app, 'integ-tests', {
+  testCases: [stack],
+});
+
+integTest.assertions.awsApiCall('Kinesis', 'putRecord', {
+  StreamARN: sourceStream.streamArn,
+  Data: 'testData123',
+  PartitionKey: '1',
+});
+
+const s3ApiCall = integTest.assertions.awsApiCall('S3', 'listObjectsV2', {
+  Bucket: bucket.bucketName,
+  MaxKeys: 1,
+}).expect(ExpectedResult.objectLike({
+  KeyCount: 1,
+})).waitForAssertions({
+  interval: cdk.Duration.seconds(30),
+  totalTimeout: cdk.Duration.minutes(10),
+});
+
+if (s3ApiCall instanceof AwsApiCall && s3ApiCall.waiterProvider) {
```

Comment:
Why do we need this `if` condition? I am not clear because `s3ApiCall` is always assigned the same value.

Another way to ask my question: When would this `if` condition evaluate to false?